packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -178,6 +189,12 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         this.node.tabIndex = 0;
     }
 
+    protected nodesWithActiveIndentGuideline = new Set<String>();
```

Comment:
Can we avoid new state and derive date in render functions only for visible rows?